src/App.tsx

```diff
@@ -1,32 +1,59 @@
-import { useEffect } from "react";
-import { Route, Routes } from "react-router-dom";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "./config/firebase-config";
+
+import { useEffect, useState } from "react";
+import { Navigate, Route, Routes } from "react-router-dom";
 import "./App.scss";
 import { Header } from "./common/components/header/header";
 import { LoginStoreImplementation } from "./features/authentication/login/store/LoginStore";
+import { ForgotPassword } from "./pages/authentication/forgot-password-page/forgot-password";
 import { Login } from "./pages/authentication/login-page/login";
 import { Register } from "./pages/authentication/register-page/register";
 import { HomePage } from "./pages/home-page/home-page";
+import { LandingPage } from "./pages/landing-page/landing-page";
+import PrivateRoute, {
+  ProtectedRouteProps,
+} from "./pages/privateRoute/private-route";
 
 interface AppProps {
   loginStore: LoginStoreImplementation;
 }
 
 export const App: React.FC<AppProps> = ({ loginStore }) => {
+  const [authenticated, setAuthenticated] = useState(true);
+
+  const defaultProtectedRouteProps: Omit<ProtectedRouteProps, "component"> = {
+    isAuthenticated: authenticated,
+    authenticationPath: "/login",
+  };
+
   useEffect(() => {
-    console.log(loginStore.login.authenticated);
-    console.log(loginStore.login.user);
+    const isAuth = JSON.parse(localStorage.getItem("authenticated") || "false");
+
+    if (isAuth === true) {
```

Comment:
There is no need to check the isAuth variable. This if-else block could be simplified to just update directly the state `setAuthenticated(isAuth);` .